crawler_mcp/crawlers/optimized/processing/url_discovery.py

```
@@ -0,0 +1,1150 @@
+                tasks.append(self._discover_from_robots(base_url))
+                source_labels.append("robots_sitemaps")
+            tasks.append(self._discover_from_common_locations(base_url))
+            source_labels.append("common_locations")
+
+            self.logger.info(
+                f"📡 Querying {len(tasks)} discovery sources: {', '.join(source_labels)}"
+            )
+
+            results = await asyncio.gather(*tasks, return_exceptions=True)
+
+            for i, (result, label) in enumerate(
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)